components/js/SlickGrid.ts

```diff
@@ -126,15 +107,22 @@ export class SlickGrid implements OnChanges, OnInit, OnDestroy, AfterViewInit {
     @Input() enableAsyncPostRender: boolean = false;
     @Input() selectionModel: string = '';
     @Input() plugins: string[] = [];
+    @Input() enableEditing: boolean = false;
+    @Input() topRowNumber: number;
+
+    @Input() isColumnEditable: (column: number) => boolean;
+    @Input() isCellEditValid: (row: number, column: number, newValue: any) => boolean;
 
     @Output() loadFinished: EventEmitter<void> = new EventEmitter<void>();
-    @Output() cellChanged: EventEmitter<{column: string, row: number, newValue: any}> = new EventEmitter<{column: string, row: number, newValue: any}>();
     @Output() editingFinished: EventEmitter<any> = new EventEmitter();
     @Output() contextMenu: EventEmitter<{x: number, y: number}> = new EventEmitter<{x: number, y: number}>();
-
-    @Input() topRowNumber: number;
     @Output() topRowNumberChange: EventEmitter<number> = new EventEmitter<number>();
 
+    @Output() cellEditBegin: EventEmitter<{row: number, column: number }> = new EventEmitter<{row: number, column: number}>();
+    @Output() cellEditTryCommit: EventEmitter<{row: number, column: number, newValue: any}> = new EventEmitter<{row: number, column: number, newValue: any}>();
```

Comment:
This isn't really a Try action. It's an event, so probably cellEditExit or cellEditCommit is fine.